src/charm.py

```
@@ -481,134 +43,61 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
-                    relation.app,
-                )
-                continue
-            dedup_ingress_relations.append(relation)
-        return tuple(nginx_route_relations + dedup_ingress_relations)
-
-    @property
-    def _all_config_or_relations(self) -> Any:
-        """Get all configuration and relation data.
+    def _get_ingress_options(self) -> Optional[IngressOption]:
+        """Retrieve the ingress options based on established relations.
 
         Returns:
-            All configuration and relation data.
-        """
-        all_relations = self._deduped_relations()
-        if not all_relations:
-            all_relations = (None,)  # type: ignore[assignment]
-        multiple_rels = self._multiple_relations
-        return [
-            _ConfigOrRelation(self.model, self.config, relation, multiple_rels)
-            for relation in all_relations
-        ]
+            IngressOption instance based on the relation data, or None if no valid relation exists.
 
-    @property
-    def _multiple_relations(self) -> bool:
-        """Check if we're related to multiple applications.
-
-        Returns:
-            if we're related to multiple applications or not.
-        """
-        return len(self._deduped_relations()) > 1
+        Raises:
+            InvalidIngressOptionError: If both "nginx-route" and "ingress" relations are present
+                or some options are invalid.
+        """
+        nginx_route_relations = self.model.relations["nginx-route"]
+        ingress_relations = self.model.relations["ingress"]
+        if nginx_route_relations and ingress_relations:
+            raise InvalidIngressOptionError(
+                "nginx-ingress-integrator cannot establish more than one relation at a time"
+            )
+        if nginx_route_relations:
+            relation = nginx_route_relations[0]
+        elif ingress_relations:
+            relation = ingress_relations[0]
+        else:
+            return None
```

This should raise an error instead of returning None